receptorctl/noxfile.py

```diff
@@ -28,12 +29,44 @@ def install(session: nox.Session, *args, req: str, **kwargs):
     session.install("-r", requirements_directory / f"{req}.in", *args, **kwargs)
 
 
+def version(session: nox.Session):
+    """
+    Create a .VERSION file.
+    """
+    try:
+        official_version = session.run(
+            "git",
+            "describe",
+            "--exact-match",
+            "--tags",
+            external=True,
+            stderr=open(os.devnull, "w"),
+        )
+    except nox.command.CommandFailed:
+        official_version = None
+
+    if not official_version:
+        print("Current commit not tagged. Using closest annotated tag instead.")
+        tag = session.run(
+            "git", "describe", "--tags", "--always", silent=True, external=True
+        )
+        rev = session.run(
+            "git", "rev-parse", "--short", "HEAD", silent=True, external=True
+        )
+        version = tag.split("-")[0] + "+" + rev
+
+    f = open(".VERSION", "w")
+    f.write(version)
+    f.close()
```

Comment:
It's safer to use a context manager with resources. So it'd be
```suggestion
    with open(".VERSION", "w") as f:
        f.write(version)
```

However, it's even better to use the higher-level APIs that hide all the complexity away:
```suggestion
    Path(".VERSION").write_text(version)
```

Reply:
Ah, I actually did use the context manager with resources initially but changed it because I thought maybe it was overly complex.

Did not know about `write_text()`, awesome.